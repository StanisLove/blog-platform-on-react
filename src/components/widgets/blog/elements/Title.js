import React from 'react';

const Title = ({title}) => (
  <h2>{title}</h2>
);

Title.propTypes = {
  title: React.PropTypes.string
}

Title.defaultProps = {
  title: 'Post'
}

export default Title;
