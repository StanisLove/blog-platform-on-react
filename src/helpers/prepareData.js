import { map } from 'lodash/collection';
import { compact } from 'lodash/array';
import { parse } from 'qs';

export default function(store, state) {
  const { location, params, routes } = state;
  const query = parse(location.search.substr(1));

  const prepareDataFns = compact(map(routes, route => route.prepareData));

  map(prepareDataFns,
    prepareData => prepareData(store, query, params, location)
  );
}
