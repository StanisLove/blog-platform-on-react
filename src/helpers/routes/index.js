export const postsPath = (id = ':id') => `/posts/${id}`;
export * from './About';
