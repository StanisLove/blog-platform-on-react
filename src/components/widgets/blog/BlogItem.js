import React from 'react';
import PropTypes from 'prop-types';
import { Item, Image, Grid } from 'semantic-ui-react';
import { Link } from 'react-router';

import Title from './elements/Title';
import TextBox from './elements/TextBox';
import Meta from './Meta';
import { postsPath } from 'helpers/routes';
import CommentForm from './CommentForm';
import CommentList from './CommentList';

const BlogItem = ({item, comments, postLiked}) => (
  <Item>
    <Grid columns={2} relaxed >
      <Grid.Column>
        <Link to={postsPath(item.id)}>
          <Image {...item.image} floated='left' />
          <Item.Header>
            <Title title={item.title} />
          </Item.Header>
        </Link>
        <TextBox text={item.text} />
        <Meta meta={item.meta} postLiked={postLiked} />
      </Grid.Column>
      <Grid.Column>
        <Title title='Comments:' />
        <CommentList comments={comments} />
      </Grid.Column>
    </Grid>
    <CommentForm postId={item.id} form={`addComment-${item.id}`} />
  </Item>
);

BlogItem.propTypes = {
  item: PropTypes.shape({
    title: PropTypes.string,
    image: PropTypes.shape({
      src: PropTypes.string,
      alt: PropTypes.string
    }),
    text: PropTypes.string,
    meta: PropTypes.shape({
      author: PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string
      }),
      postDates: PropTypes.shape({
        createdAt: PropTypes.string,
        updatedAt: PropTypes.string
      }),
      likeCont: PropTypes.number
    })
  }),
  postLiked: PropTypes.func,
  comments: PropTypes.array
};

BlogItem.defaultProps = {
  item: {
    image: {
      src: 'http://proxyprivat.com/images/noimage.jpeg',
      alt: 'No image'
    },
    text: 'You can add some text',
    meta: {
      author: { firstName: 'Anonymous', lastName: 'Anonymous' },
      postDates: { createdAt: 'undefined', updatedAt: 'undefined' },
      likeCount: 0
    }
  },
  comments: []
};

export default BlogItem;
