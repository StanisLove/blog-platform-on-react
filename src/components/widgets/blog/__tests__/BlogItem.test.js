import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { fakeStore } from 'helpers/fakeStore';

import BlogItem from '../BlogItem';
import Title from '../elements/Title';

describe('BlogItem', () => {
  it('returns without crashing', () => {
    const div = document.createElement('div');
    const store = fakeStore({});
    ReactDOM.render(
      <Provider store={store}>
        <BlogItem />
      </Provider>,
      div
    );
  });
});

describe('render', () => {
  const blogItemProps = { title: 'Some title', id: 5 };
  const blogItem = shallow(<BlogItem item={blogItemProps} />);

  it('should render link', () => {
    const path = `/posts/${blogItemProps.id}`;
    const link = blogItem.find('Link').first();

    expect(link.props().to).toEqual(path);
  });

  it('should render title', () => {
    const title = <Title title={blogItemProps.title} />;

    expect(blogItem.contains(title)).toEqual(true);
  });

  it('render dumb component', () => {
    expect(blogItem).toMatchSnapshot();
  });
});
