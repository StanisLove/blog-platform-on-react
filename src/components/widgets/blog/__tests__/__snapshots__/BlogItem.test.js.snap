// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render render dumb component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Item>
    <Grid
        columns={2}
        relaxed={true}
    >
        <GridColumn>
            <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/posts/5"
            >
                <Image
                    as="img"
                    floated="left"
                    ui={true}
                />
                <ItemHeader>
                    <Title
                        title="Some title"
                    />
                </ItemHeader>
            </Link>
            <TextBox
                text="You can add some text"
            />
            <Meta
                meta={
                    Object {
                        "author": Object {
                          "firstName": "Anonymous",
                          "lastName": "Anonymous",
                        },
                        "likeCount": 0,
                        "postDates": Object {
                          "createdAt": "undefined",
                          "updatedAt": "undefined",
                        },
                      }
                }
                postLiked={undefined}
            />
        </GridColumn>
        <GridColumn>
            <Title
                title="Comments:"
            />
            <CommentList
                comments={Array []}
            />
        </GridColumn>
    </Grid>
    <ReduxForm
        form="addComment-5"
        postId={5}
    />
</Item>,
  "nodes": Array [
    <Item>
      <Grid
            columns={2}
            relaxed={true}
      >
            <GridColumn>
                  <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/posts/5"
                  >
                        <Image
                              as="img"
                              floated="left"
                              ui={true}
                        />
                        <ItemHeader>
                              <Title
                                    title="Some title"
                              />
                        </ItemHeader>
                  </Link>
                  <TextBox
                        text="You can add some text"
                  />
                  <Meta
                        meta={
                              Object {
                                    "author": Object {
                                      "firstName": "Anonymous",
                                      "lastName": "Anonymous",
                                    },
                                    "likeCount": 0,
                                    "postDates": Object {
                                      "createdAt": "undefined",
                                      "updatedAt": "undefined",
                                    },
                                  }
                        }
                        postLiked={undefined}
                  />
            </GridColumn>
            <GridColumn>
                  <Title
                        title="Comments:"
                  />
                  <CommentList
                        comments={Array []}
                  />
            </GridColumn>
      </Grid>
      <ReduxForm
            form="addComment-5"
            postId={5}
      />
</Item>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogItem
        comments={Array []}
        item={
                Object {
                        "id": 5,
                        "title": "Some title",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "comments": Array [],
          "item": Object {
            "id": 5,
            "title": "Some title",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Item>
          <Grid
                    columns={2}
                    relaxed={true}
          >
                    <GridColumn>
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/5"
                              >
                                        <Image
                                                  as="img"
                                                  floated="left"
                                                  ui={true}
                                        />
                                        <ItemHeader>
                                                  <Title
                                                            title="Some title"
                                                  />
                                        </ItemHeader>
                              </Link>
                              <TextBox
                                        text="You can add some text"
                              />
                              <Meta
                                        meta={
                                                  Object {
                                                            "author": Object {
                                                              "firstName": "Anonymous",
                                                              "lastName": "Anonymous",
                                                            },
                                                            "likeCount": 0,
                                                            "postDates": Object {
                                                              "createdAt": "undefined",
                                                              "updatedAt": "undefined",
                                                            },
                                                          }
                                        }
                                        postLiked={undefined}
                              />
                    </GridColumn>
                    <GridColumn>
                              <Title
                                        title="Comments:"
                              />
                              <CommentList
                                        comments={Array []}
                              />
                    </GridColumn>
          </Grid>
          <ReduxForm
                    form="addComment-5"
                    postId={5}
          />
</Item>,
        "_debugID": 2,
        "_renderedOutput": <Item>
          <Grid
                    columns={2}
                    relaxed={true}
          >
                    <GridColumn>
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/5"
                              >
                                        <Image
                                                  as="img"
                                                  floated="left"
                                                  ui={true}
                                        />
                                        <ItemHeader>
                                                  <Title
                                                            title="Some title"
                                                  />
                                        </ItemHeader>
                              </Link>
                              <TextBox
                                        text="You can add some text"
                              />
                              <Meta
                                        meta={
                                                  Object {
                                                            "author": Object {
                                                              "firstName": "Anonymous",
                                                              "lastName": "Anonymous",
                                                            },
                                                            "likeCount": 0,
                                                            "postDates": Object {
                                                              "createdAt": "undefined",
                                                              "updatedAt": "undefined",
                                                            },
                                                          }
                                        }
                                        postLiked={undefined}
                              />
                    </GridColumn>
                    <GridColumn>
                              <Title
                                        title="Comments:"
                              />
                              <CommentList
                                        comments={Array []}
                              />
                    </GridColumn>
          </Grid>
          <ReduxForm
                    form="addComment-5"
                    postId={5}
          />
</Item>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    comments={Array []}
    item={
        Object {
            "id": 5,
            "title": "Some title",
          }
    }
/>,
}
`;
